mongodb/migrations/migrations.go

```diff
@@ -180,13 +181,63 @@ var m004 = migrate.Migration{
 	},
 }
 
+var m005 = migrate.Migration{
+	Version:     5,
+	Description: "rename any bucket archive job_status to status",
+	Up: func(db *mongo.Database) error {
+		log.Info("migrating 005 up")
+		ctx, cancel := context.WithTimeout(context.Background(), migrateTimeout)
+		defer cancel()
+		if _, err := db.Collection("bucketarchives").UpdateMany(ctx, bson.M{}, bson.M{
+			"$rename": bson.M{"archives.current.job_status": "archives.current.status"},
+		}); err != nil {
+			return err
+		}
+		if _, err := db.Collection("bucketarchives").Aggregate(ctx, mongo.Pipeline{
+			bson.D{primitive.E{Key: "$addFields", Value: bson.M{
+				"archives.history": bson.M{
+					"$map": bson.M{
+						"input": "$archives.history",
+						"as":    "archive",
+						"in": bson.M{
+							"$mergeObjects": bson.A{
+								bson.M{"status": "$$archive.job_status"},
+								bson.M{
+									"$arrayToObject": bson.M{
+										"$filter": bson.M{
+											"input": bson.M{"$objectToArray": "$$archive"},
+											"as":    "archive",
+											"cond":  bson.M{"$ne": bson.A{"$$archive.k", "job_status"}},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}}},
+			bson.D{
+				primitive.E{Key: "$out", Value: "bucketarchives"},
+			},
+		}); err != nil {
+			return err
+		}
```

Comment:
You can't use `$rename` on objects that are nested in arrays, so you have to use `Aggregate`, writing the results back to the collection. This is pretty next level stuff, I'm not going to try to explain it all because it was some copy/paste/modify. I leave it to the very thorough migration test to verify this does indeed work.